geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/ManagementControllerAdvice.java

```diff
@@ -33,9 +45,47 @@
 import org.apache.geode.security.NotAuthorizedException;
 
 @ControllerAdvice
-public class ManagementControllerAdvice {
+public class ManagementControllerAdvice implements ResponseBodyAdvice<Object> {
   private static final Logger logger = LogService.getLogger();
 
+  @Autowired
+  private Jackson2ObjectMapperFactoryBean objectMapperFactory;
+
+  @Override
+  public boolean supports(MethodParameter returnType, Class converterType) {
+    return AbstractJackson2HttpMessageConverter.class.isAssignableFrom(converterType);
+  }
+
+  @Override
+  public Object beforeBodyWrite(Object body, MethodParameter returnType,
+      MediaType selectedContentType, Class selectedConverterType,
+      ServerHttpRequest request, ServerHttpResponse response) {
+    boolean includeClass = request.getHeaders().containsKey("X-Include-Class");
+
+    try {
+      ObjectMapper objectMapper = objectMapperFactory.getObject();
+      String json = objectMapper.writeValueAsString(body);
+      if (!includeClass) {
+        json = removeClassFromJsonText(json);
+      }
+      response.getHeaders().add(HttpHeaders.CONTENT_TYPE,
+          "application/json; charset=" + Charset.defaultCharset().toString());
+      response.getBody().write(json.getBytes());
+      response.close();
+      return null;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static String removeClassFromJsonText(String json) {
```

Comment:
Add some unit tests for removeClassFromJsonText to make sure it does what is expected.

Reply:
added